### PR TITLE
App: Fix saving required relational fields

### LINKED ISSUE #17897: Many to many field is  required not honoured when saving
### Describe the Bug

suppose I have  a many-to-many field in my collection in that I have set the value of is required to true and then I delete all the values of the m2m field and try to save it  get saved without giving any error 
and then i open the same item in the collection and try to add the value in the m2m field and try to save it always give the error the following xx m2m field cannot be nullI

### To Reproduce

1)create any collection
2)add m2m value with the required set to true
3)add the item to the collection and save it
4)open the same item delete all the values from m2m field and save it get saved ideally it should throw an error
5)open the same item again and try to repopulate the m2m value now it is throwing error xx m2m filed can't be null and it is stuck there
 version : latest

### Hosting Strategy

Self-Hosted (Docker Image)

### PR DESCRIPTION
**Pull request by joselcvarela**
## Description
When trying to save a required relationship an error is shown saying `<Field> cant' be null`.
After some investigation I found that when we `save` an item the following function is called:
https://github.com/directus/directus/blob/6a54e24bbafd63aa8b90038bda0fa14b95046004/app/src/composables/use-item.ts#L123-L127

As we can see, we are trying to merge `defaultValues.value`, `item.value` and `edits.value`.
Everything works as expected until we find a relationship. That's because the value of a relationship on `item.value` is an Array while on `edits.value` is an Object like `{create:...,update:...,delete:...}`. Also the result of `merge` with Array and Object is an Array with the Object properties, i.e. the indexes of the array are `create`, `update`, `delete` instead of `0`, `1`, `2` and so on. 
When it reaches here:
https://github.com/directus/directus/blob/6a54e24bbafd63aa8b90038bda0fa14b95046004/app/src/utils/validate-item.ts#L19
The `cloneDeep` does not recognize those weird indexes on array and the result is `[]` leading to the error, because the array is empty.

## Solution
Instead of using `merge`, we can just check if the next object of merge has value and if so, we just use that value. We can do so using `mergeWith`.
Since this is only to validate the payload, I believe that just overriding previous value in case the current one exists can be sufficient. 

## Preview
| Before | After |
| - | - |
| <video src="https://user-images.githubusercontent.com/14039341/226144832-c00dd2d1-e9d9-4122-b675-f0e36eb2e9c9.mov" /> | <video src="https://user-images.githubusercontent.com/14039341/226144839-99776452-a9c6-489e-be4a-23a6d9defb3e.mov" /> | 

Fixes ENG-822
Fixes #17897

## Type of Change

- [x] Bugfix
- [ ] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [ ] New / updated tests are included
- [ ] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated. PR:

### CODE REVIEW

File: app/src/composables/use-item.ts [hunk 1]
```diff
@@ -120,12 +120,20 @@ export function useItem(
...
-			merge({}, defaultValues.value, item.value, edits.value),
-			fieldsWithPermissions.value,
-			isNew.value
+		const payloadToValidate = mergeWith(
+			{},
+			defaultValues.value,
+			item.value,
+			edits.value,
+			function (from: any, to: any) {
+				if (typeof to !== undefined) {
```
Comment (Nitwel) [inline app/src/composables/use-item.ts:129]: `typeof to !== undefined` will always evaluate to true as `typeof` always returns a string.

**Review by Nitwel — CHANGES_REQUESTED** [review]
This is not a full solution as after creating an item and then deleting the relational item, the relation will become empty but while saving the item, the relational field is not empty as it still contains data on what item to delete. E.g. `{create: [], update: [], delete: [1]}`. In order to fully support this we would need the full item count in the relation from the api or in the future add a new endpoint that validates changes api side and we can then properly show validation.

File: app/src/composables/use-item.ts (hunk 1 above)
Comment (joselcvarela) [reply app/src/composables/use-item.ts]: Nice catch 🤝

**Review by Nitwel — APPROVED** [review]
I'm fine with this improvement as it's the best thing we can do for now with the problem of only sending changes on relational and validation

**Review by br41nslug — APPROVED** [review]
lgtm 🚀

**Pull request merged**